packages/blockchain/src/consensus/ethash.ts

```diff
@@ -0,0 +1,28 @@
+import { Block } from '@ethereumjs/block'
+import Ethash from '@ethereumjs/ethash'
+import Blockchain from '..'
+import { Consensus, ConsensusOptions } from './interface'
+
+/**
+ * This class encapsulates Ethash-related consensus functionality when used with the Blockchain class.
+ */
+export class EthashConsensus implements Consensus {
+  blockchain: Blockchain
+  _ethash: Ethash
+
+  constructor({ blockchain }: ConsensusOptions) {
+    this.blockchain = blockchain
+    this._ethash = new Ethash(this.blockchain.db)
+  }
+
+  async validate(block: Block): Promise<void> {
+    const valid = await this._ethash.verifyPOW(block)
```

Comment:
could we maybe add a test to add coverage for this line? bonus would be one more test to test invalid :)